src/components/modal/ModalHeaderConnected.tsx

```diff
@@ -1,9 +1,20 @@
 import {connect} from 'react-redux';
+import * as _ from 'underscore';
+import {IReactVaporState} from '../../ReactVapor';
 import {IDispatch, ReduxUtils} from '../../utils/ReduxUtils';
 import {closeModal} from './ModalActions';
-import {IModalHeaderDispatchProps, IModalHeaderOwnProps, IModalHeaderProps, ModalHeader} from './ModalHeader';
+import {
+    IModalHeaderDispatchProps, IModalHeaderOwnProps, IModalHeaderProps, IModalHeaderStateProps,
+    ModalHeader,
+} from './ModalHeader';
 
-const mapStateToProps = () => ({});
+const mapStateToProps = (state: IReactVaporState, ownProps: IModalHeaderOwnProps): IModalHeaderStateProps => {
+    const lastOpenedModal = _.last(state.openModals);
+
+    return {
+        lastOpened: ownProps.id === lastOpenedModal,

```

Comment:
you could just name this `canClose` if this is the only condition for it to close, or just use `this.props.lastOpened`   
  
```  
if (this.canClose && this.props.onClose) {  
```